unstract/connectors/src/unstract/connectors/filesystems/sftp/sftp.py

```diff
@@ -30,9 +29,13 @@ def __init__(self, settings: dict[str, Any]):
         port = int(settings.get(SettingsKey.PORT, SettingsDefault.PORT))
         username = settings.get(SettingsKey.USERNAME)
         password = settings.get(SettingsKey.PASSWORD)
+        self.directory = str(settings.get(SettingsKey.USER_DIRECTORY))
```

Comment:
@chandrasekharan-zipstack sftp has one directory for the user. All remaining directories are accessible by root.
As you can see from the docker compose file, `docker/connector-db/docker-compose-db-connector.yaml`
command: "${SFTP_USER}:${SFTP_PASS}:${SFTP_USER_ID}::${SFTP_DIR}"

`SFTP_DIR` is the directory associated with user. This will help the user use the directory associated. 
If he/she selects any other directory, then while running the workflow, we would through an error, which I have mentioned in the screenshot

Reply:
Okay as long as we've enforced this its okay